apps/condo/domains/common/components/tasks/TaskProgress.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useEffect, useState } from 'react'
+/** @jsx jsx */
+import { css, jsx } from '@emotion/react'
+import styled from '@emotion/styled'
+import { Progress, Typography, List, Row, Col } from 'antd'
+import { useIntl } from '@core/next/intl'
+import { TASK_POLL_INTERVAL, TASK_COMPLETED_STATUS } from '../../constants/tasks'
+import { colors } from '../../constants/style'
+import {
+    IClientSchema,
+    OnCompleteFunc,
+    TaskRecord,
+    ITaskTrackableItem,
+    TaskProgressTranslations,
+    CalculateProgressFunc,
+    TaskDisplayProgressValue, TASK_PROGRESS_UNKNOWN,
+} from './index'
+import { InfoCircleOutlined } from '@ant-design/icons'
+import { CheckIcon } from '../icons/Check'
+import { MinusOutlined } from '@ant-design/icons'
+import { ExpandIcon } from '../icons/ExpandIcon'
+
+const InfiniteSpinningStyle = css`
+  @keyframes rotate {
+    100% {
+        transform: rotate(1turn);
+    }
+  }
+  .ant-progress-circle {
+    animation: rotate 1.6s linear infinite;
+  }
+`
+
+type ICircularProgressProps = {
+    progress: TaskDisplayProgressValue,
+}
+
+/**
+ * Gradient-colored circle that can be displayed in two states depending on value of a `progress` prop:
+ * 1. Known progress displaying as partially filled circle
+ * 2. Unknown progress displaying as "fully" filled spinning circle
+ * "Fully" filling is implemented as 99 percents to prevent displaying Ant `Progress` component in completed state
+ */
+export const CircularProgress = ({ progress }: ICircularProgressProps) => (
+    <Progress
+        type="circle"
+        width={22}
+        strokeWidth={14}
+        strokeColor={{
+            '0%': '#4CD174',
+            '100%': '#6DB8F2',
+        }}
+        strokeLinecap="square"
+        showInfo={false}
+        percent={progress === TASK_PROGRESS_UNKNOWN ? 99 : progress}
```

Comment:
Why when task progress is unknown progress bar value sets to 99? May be set to zero?

Reply:
Read comments for `CircularProgress` component